preprocessor/src/test/java/com/azure/autorest/MockPreprocessor.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+package com.azure.autorest;
+
+import com.azure.autorest.extension.base.jsonrpc.Connection;
+import com.azure.autorest.extension.base.model.codemodel.CodeModel;
+import com.azure.autorest.extension.base.plugin.JavaSettings;
+import com.azure.autorest.preprocessor.Preprocessor;
+import com.azure.autorest.preprocessor.tranformer.Transformer;
+import org.yaml.snakeyaml.DumperOptions;
+import org.yaml.snakeyaml.LoaderOptions;
+import org.yaml.snakeyaml.Yaml;
+import org.yaml.snakeyaml.constructor.Constructor;
+import org.yaml.snakeyaml.introspector.Property;
+import org.yaml.snakeyaml.nodes.NodeTuple;
+import org.yaml.snakeyaml.nodes.Tag;
+import org.yaml.snakeyaml.representer.Representer;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.Map;
+
+public class MockPreprocessor extends Preprocessor {
+
+
+    private static final Map<String, Object> SETTINGS_MAP = new HashMap<>();
+
+    static {
+        SETTINGS_MAP.put("generate-client-interfaces", false);
+        SETTINGS_MAP.put("generate-client-as-impl", true);
+        SETTINGS_MAP.put("generate-sync-async-clients", true);
+        SETTINGS_MAP.put("add-context-parameter", true);
+        SETTINGS_MAP.put("context-client-method-parameter", true);
+        SETTINGS_MAP.put("sync-methods", "all");
```

Comment:
Do we need all these in preprocessor test?

Reply:
Maybe... not... I copied it from readme under preprocessor. I removed it and rerun the test to find it pass. 
Will remove it in the next pr.